### PR TITLE
Serialising updates

### LINKED ISSUE #26130: PoC to extend the data limit the app can handle
Tested [previously](https://www.notion.so/appsmith/Code-optimizations-around-large-data-handling-of-eval-computations-7a06dbc24ab84b69b9ee8f0849f0151d?pvs=4#3ca77e06ddaf401680d388e843f72227) our app can handle until 40000 records, post that the app begins to crash as it runs out of memory. After profiling it has been observed that the following pieces of code in the evalTree seems to be causing a spike in memory heap consumption.
- **removeFunctions**
- **santisize dataTree** JSON.parse(JSON.stringify(newDataTree))
- **Klona** clones of the dataTree
We will try to fix these pieces of code and test what is the upper bound of data our app can handle.

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description

- In this PR the updates are directly serialised and sent to the main thread
- These updates are subsequently parsed by the main thread.
- Optimised code around makeEntityConfigsAsObjProperties.
- Observed the data limit our app can handle roughly doubled with these changes.

#### PR fixes following issue(s)
Fixes #26130

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/workers/Evaluation/evalTreeWithChanges.ts [hunk 1]
```diff
@@ -82,8 +84,19 @@ export function evalTreeWithChanges(
     dataTree,
     dataTreeEvaluator,
   );
+
+  let serUpdates = "[]";
+  try {
+    serUpdates = serialiseUpdates(updates);
+  } catch (error) {
+    errors.push({
```
Comment (arunvjn) [inline app/client/src/workers/Evaluation/evalTreeWithChanges.ts:92]: Can you paste an anonymous function into a widget field that doesn't contain a validation config against it and see what happens ?

Comment (arunvjn) [reply app/client/src/workers/Evaluation/evalTreeWithChanges.ts:92]: You could experiment with headers field in an API

**Comment by laveena-en** [comment]
@vsvamsi1 One of the widgets which contains large data just refused to load. It isn’t intermittent either. Loom: https://www.loom.com/share/60474f19f08a4a40826c9cbd48c8dd94?sid=d8ac488b-a103-436f-8897-66359627be58

[ Application](https://ce-26214.dp.appsmith.com/app/my-first-application/page1-64ddd87ad050df42a2bfda38?embed=true) Can you please check?

**Comment by vsvamsi1** [comment]
> @vsvamsi1 One of the widgets which contains large data just refused to load. It isn’t intermittent either. Loom: https://www.loom.com/share/60474f19f08a4a40826c9cbd48c8dd94?sid=d8ac488b-a103-436f-8897-66359627be58
> 
> [ Application](https://ce-26214.dp.appsmith.com/app/my-first-application/page1-64ddd87ad050df42a2bfda38?embed=true) Can you please check?

Sure @laveena-en I will study this issue

**Pull request closed without merging**